bazel/foreign_cc/BUILD

```diff
@@ -0,0 +1,39 @@
+licenses(["notice"])  # Apache 2
+
+load("//bazel:envoy_build_system.bzl", "envoy_package")
+load("@rules_foreign_cc//tools/build_defs:cmake.bzl", "cmake_external")
+load("@foreign_cc_platform_utils//:tools.bzl", "NINJA_COMMAND", "NINJA_DEP")
+
+envoy_package()
+
+cmake_external(
+    name = "ares",
+    cache_entries = {
+        "CARES_SHARED": "no",
+        "CARES_STATIC": "on",
+    },
+    cmake_options = ["-GNinja"],
+    lib_source = "@com_github_c_ares_c_ares//:all",
+    make_commands = [
+        NINJA_COMMAND,
+        NINJA_COMMAND + " install",
+    ],
+    static_libraries = ["libcares.a"],
+    tools_deps = NINJA_DEP,
+)
+
+cmake_external(
```

Comment:
@sesmith177 any chance you can try out this PR and let me know if it does the right thing on Windows?

Reply:
we can take a look at it when we get a chance. One thing we noticed is that the names of the static libraries are different on Windows and Linux ( `cares.lib` and `nghttp2.lib` vs `libcares.a` and `libnghttp2.a`)

Reply:
Hi, here's my modified cares build for Windows (I just completed it, will update the examples):

cmake_external(
    name = "cares",
    cache_entries = {
        "CARES_SHARED": "no",
        "CARES_STATIC": "on",
    },
    defines = ["CARES_STATICLIB"],
    cmake_options = ["-GNinja"],
    lib_source = "@cares//:all",
    make_commands = [
        "ninja",
        "ninja install",
    ],
)

1. Since on Wndows cares.lib is created, "static_libraries" can be omitted (cares.lib is the calculated default)
2.   defines = ["CARES_STATICLIB"], is very important, for some reason CMake script does not set this up because of corresponding CMake variable
3. I used the preinstalled ninja, seems one should not build it on Windows

Hope it helps. Will be happy to answer any questions.